tracer/src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -396,21 +397,39 @@ SpanBuffer SwapBuffers()
             }
 
             RunSpanSampler(spans);
-
+            int? chunkSamplingPriority = null;
             if (CanComputeStats)
             {
                 spans = _statsAggregator?.ProcessTrace(spans) ?? spans;
                 bool shouldSendTrace = _statsAggregator?.ShouldKeepTrace(spans) ?? true;
                 _statsAggregator?.AddRange(spans);
+                var singleSpanSamplingSpans = new List<Span>(); // TODO maybe we can store this from above?
+
+                for (var i = spans.Offset; i < spans.Count; i++)
+                {
+                    if (spans.Array![i].GetMetric(Metrics.SingleSpanSampling.SamplingMechanism) is not null)
+                    {
+                        singleSpanSamplingSpans.Add(spans.Array![i]);
+                    }
+                }
 
                 // If stats computation determined that we can drop the P0 Trace,
                 // skip all other processing
-                if (!shouldSendTrace)
+                if (!shouldSendTrace && singleSpanSamplingSpans.Count == 0)
                 {
                     Interlocked.Increment(ref _droppedP0Traces);
                     Interlocked.Add(ref _droppedP0Spans, spans.Count);
                     return;
                 }
+                else if (!shouldSendTrace && singleSpanSamplingSpans.Count > 0)
```

Comment:
Maybe just me, but I would prefer to nest these `if` conditions like this:
```
if (!shouldSendTrace)
{
    if (singleSpanSamplingSpans.Count == 0)
    {
        // ...
    }
    else // singleSpanSamplingSpans.Count > 0
    {
        // ...
    }
}
```

Level of care low.